test/test_config.rb

```diff
@@ -687,6 +687,16 @@ def test_config_loads_workers_from_env
     end
   end
 
+  def test_config_sets_workers_to_processor_count_with_auto
+    require "concurrent/utility/processor_counter" # defines Concurrent to allow stubbing
+    ::Concurrent.stub :available_processor_count, 42 do
```

Comment:
@MSP-Greg had concerns about loading `concurrent-ruby` in the main test process, https://github.com/puma/puma/issues/3437#issuecomment-2278105545

I agree with that. Not sure exactly how to accomplish it, one way is to launch a sub-process with `BUNDLE_GEMFILE` set for it, pointing to a Gemfile with `concurrent-ruby`. The assert becomes trickier, maybe need to output something in a HTTP response? Maybe there is some Puma tests already doing something similar?

Reply:
On first glance, maybe I can write an integration test similar to `test_url_map.rb`. I'll look into it 👍 . Maybe this is also a good way to test the warning message if `concurrent-ruby` is missing.